### PR TITLE
Add an RemoveFromBuffer method for test

### PR DESCRIPTION
**Pull request by ekexium**
Removing a record from the membuffer can be useful for test, e.g. in pingcap/tidb#26833.

### CODE REVIEW

File: internal/unionstore/memdb.go [hunk 1]
```diff
@@ -833,3 +833,13 @@ func (n *memdbNode) getKeyFlags() kv.KeyFlags {
 func (n *memdbNode) setKeyFlags(f kv.KeyFlags) {
 	n.flags = (^nodeFlagsMask & n.flags) | uint16(f)
 }
+
+// UnsafeRemoveRecord removes a record from the mem buffer. It should be only used for test.
+func (db *MemDB) UnsafeRemoveRecord(key []byte) {
```
Comment (MyonKeminta) [inline internal/unionstore/memdb.go:838]: After some consideration, I think `RemoveFromBuffer` should be a more clear and understandable name.
By the way, you can remove the `DeleteKey` function in memdb_test.go and change its usages to this new one.

**Comment by ekexium** [comment]
I just found it's already in #311 🥲

**Comment by MyonKeminta** [comment]
> I just found it's already in #311 🥲

I think let's adopt this one 😂

**Comment by MyonKeminta** [comment]
PTAL @cfzjywxk 
Can someone add you to the Github organization? We can't add you to the reviewers of PRs 😂

**Review by MyonKeminta — APPROVED** [review]

**Review by cfzjywxk — APPROVED** [review]
LGTM

**Pull request merged**